flocker/acceptance/endtoend/test_dockerplugin.py

```diff
@@ -48,6 +50,25 @@ def get_path(name):
         return Client(base_url="https://{}:{}".format(address, DOCKER_PORT),
                       tls=tls, timeout=100)
 
+    def restart_docker(self, address):
+        """
+        Restart the Docker daemon on the specified address.
+
+        :param bytes address: The public IP of the node on which Docker will
+            be restarted.
+        """
+        try:
+            distro = run_ssh_command(
+                b"root", address, ["python", "-m", "platform"])
+            distro = distro.output.lower()
+            if 'ubuntu' in distro:
+                command = ["service", "docker", "restart"]
+            else:
+                command = ["systemctl", "restart", "docker"]
+            run_ssh_command(b"root", address, command)
```

Comment:
As far as I know, we don't have root SSH for our cloud instances, but the only ways I can think of to restart the Docker daemon on a node is either SSH or an interface to restart the machine. I'd be glad to know someone else's thoughts on what best to do here - we may want to do something like set the SSH username (and perhaps a key path) in the environment..